features/unattached_commands.feature

```diff
@@ -0,0 +1,63 @@
+Feature: Command behaviour when unattached
+
+    Scenario: Unattached detach in a trusty lxd container
+        Given a trusty lxd container
+          And ubuntu-advantage-client is installed
+        When I run `ua detach`
+        Then I will see the following on stderr:
```

Comment:
I see the error? (negative)
I see the output? (positive)

Reply:
ok, but if you are saying "see the following on stderr" does that mean you are actually checking stderr and not stdout? I guess I'm wondering do we need to make the distinction.

Reply:
Yep, each one checks its own thing:

Reply:
(Previous comment was edited, in case you're only following along via email.)

I do think maintaining a distinction between the two is important, because it does make a substantial difference to the way that users interact with the command programmatically.

That said, we could also add a more general "Then I will see the following output", but the implementation could be a little tricky, because we would (probably) have to redirect stderr to stdout (easy) which would mean that the "I will see the following on (stdout|stderr)" would no longer function correctly. Not impossible to sort out, for sure, but a wrinkle nonetheless.

Reply:
ok you have resolved my concerns :) having both makes sense

Reply:
Thanks!